vulkan_generator/handle_remapper/generator.py

```diff
@@ -0,0 +1,299 @@
+        {map_name}[captureHandle] = replayHandle;"""
+    )
+
+def dispatchable_handle_remove_code(handle: str) -> str:
+    map_name = handle_map_name(handle)
+    return dedent(f"""
+        if({map_name}.find(captureHandle) == {map_name}.end()) throw RemoveNonExistantHandleException();
+        {map_name}.erase(captureHandle);"""
+    )
+
+def dispatchable_handle_remap_code(handle: str) -> str:
+    map_name = handle_map_name(handle)
+    return dedent(f"""
+        if({map_name}.find(captureHandle) == {map_name}.end()) throw RemapNonExistantHandleException();
+        return {map_name}[captureHandle];"""
+    )
+
+def non_dispatchable_handle_add_code(handle: str) -> str:
+    map_name = handle_map_name(handle)
+    map_count_name = handle_count_map_name(handle)
+    return dedent(f"""
+        auto map_iter = {map_name}.find(captureHandle);
+        auto map_count_iter = {map_count_name}.find(captureHandle);
+
+        if(map_iter == {map_name}.end()) {{
+            if(map_count_iter != {map_count_name}.end()) throw InternalConsistencyException();
+        }}
+        else {{
+            if(map_count_iter == {map_count_name}.end() || map_count_iter->second <= 0) throw InternalConsistencyException();
+            if(map_iter->second != replayHandle) throw NonDispatchableHandleRedefinitionException();
+        }}
+
+        {map_name}[captureHandle] = replayHandle;
+        {map_count_name}[captureHandle] = {map_count_name}[captureHandle] +1;"""
+    )
+
+def non_dispatchable_handle_remove_code(handle: str) -> str:
+    map_name = handle_map_name(handle)
+    map_count_name = handle_count_map_name(handle)
+    return dedent(f"""
+        auto map_iter = {map_name}.find(captureHandle);
+        auto map_count_iter = {map_count_name}.find(captureHandle);
+
+        if(map_iter == {map_name}.end() || map_count_iter == {map_count_name}.end()) {{
+            throw RemoveNonExistantHandleException();
+        }}
+        else {{
+            if(map_count_iter->second <= 0) throw InternalConsistencyException();
+        }}
+
+        if(({map_count_name}[captureHandle] = {map_count_name}[captureHandle] -1) <= 0){{
+            {map_name}.erase(map_iter);
+            {map_count_name}.erase(map_count_iter);
+        }}"""
+    )
+
+def non_dispatchable_handle_remap_code(handle: str) -> str:
+    map_name = handle_map_name(handle)
+    map_count_name = handle_count_map_name(handle)
+    return dedent(f"""
+        auto map_iter = {map_name}.find(captureHandle);
+        auto map_count_iter = {map_count_name}.find(captureHandle);
+
+        if(map_iter == {map_name}.end() || map_count_iter == {map_count_name}.end()) {{
+            throw RemapNonExistantHandleException();
+        }}
+        else {{
+            if(map_count_iter->second <= 0) throw InternalConsistencyException();
+        }}
+
+        return map_iter->second;"""
+    )
+
+def handle_add_code(handle: str) -> str:
```

Comment:
A thought:

As we will always have these three functions(and maybe more in the future) it might be useful to return one string that has all functions instead of three.

e.g.
```
def generate_handle_code(handle: str) -> str:
if dispatchable:
  # this can be fstring if necessary
   return dispatchable_handle_add_code + dispatchable_handle_remove_code +...
```

Reply:
I strongly disagree that we should have one function to generate 3 separate pieces of code. One function, one purpose.